### ISSUE TITLE
Uncertain if allow_blank is handled properly.

### ISSUE THREAD

**Issue by tomciopp** [bogdan/datagrid#138]
I am using datagrid to filter records by their id, and I am a bit confused by how datagrid is handling a request of mine, although I am not sure it is *incorrect* based on the phrasing of the docs. Here is my example, and what I expect to happen.

    class TicketFilter 
      include Datagrid

      scope { Ticket.order('name ASC') }

      filter(:id, multiple: true, allow_blank: true)
    end

Here are the different cases that I need to handle:

    1.) TicketFilter.new.assets 
    Expectation => returns all records in ascending order
    Reality => returns an empty array

    2.) TicketFilter.new(id: [1, 2, etc.]).assets
    Expectation => returns Ticket:1, 2, etc.
    Reality => returns Ticket: 1, 2, etc.

    3.) TicketFilter.new(id: []).assets
    Expectation => returns an empty array
    Reality => returns an empty array

The issue I am having is that even though I do not specify the id as an option when initializing the Filter, the id filter is called when retrieving the assets. (Scenario 1 & 3 are functionally equivalent)

I have implemented a hacky method of getting datagrid to do what I want, but I was wondering if the following code did not make sense to you as well. If `allow_blank?`'s behavior cannot be changed can you add another option like :run_if_present? which automatically runs regardless of the value it is given.

**Comment by bogdan** [comment]
The correct configuration on your case should be: 

``` ruby
filter(:id, multiple: true, allow_nil: false, allow_blank: true)
```

But still this 3 options work incorrectly when combined all together.

There is fixed in master: https://github.com/bogdan/datagrid/commit/96b1a1795844567ce9bc696e836ab6b6b291675b

Please try to use datagrid from master and confirm that it fixes your problem and I will make a patch release.

Just FYI: if you feel that options don't work in some case you are still able to control everything completely manually:

``` ruby
filter(:id, multiple: true, allow_nil: true, allow_blank: true) do |value, scope|
  if value.nil?
    scope
  elsif value.empty?
    scope.none
  else
    scope.where(id: value)
  end
end
```

**Comment by tomciopp** [comment]
Ref: 96b1a1795844567ce9bc696e836ab6b6b291675b fixes the issues that I was having :100:

**Comment by bogdan** [comment]
Released version 1.3.4
